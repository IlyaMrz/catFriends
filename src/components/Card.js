import React from 'react';

const Card = ({ name, email, id, city }) => {
    return (
        <div className='bg-light-green tc dib br3 pa3 ma2 grow bw2 shadow-3'>
            <img alt='123' src={`https://robohash.org/set_set4/${id * 4}?size=200x200`} />
            <div>
                <h2>{name}</h2>
                <p>{email}</p>
                <p>City: {city}</p>
            </div>
        </div>
    );
}

export default Card;