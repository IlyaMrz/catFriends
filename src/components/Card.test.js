import { shallow, mount, render } from 'enzyme';
import React from 'react';
import Card from './Card';
import '../setupTest';
// https://backbencher.dev/blog/empty-shallowwrapper-snapshot-jest-enzyme
import { create } from "react-test-renderer";

const cardd = create(<Card />)


it("expect to render Card component",()=> {
    expect(cardd.toJSON()).toMatchSnapshot()
})

// it("expect to render Card component",()=> {
//     expect(shallow(<Card />)).toMatchSnapshot()
// })
