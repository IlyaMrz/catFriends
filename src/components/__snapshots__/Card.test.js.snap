// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
<div
  className="bg-light-green tc dib br3 pa3 ma2 grow bw2 shadow-3"
>
  <img
    alt="123"
    src="https://robohash.org/set_set4/NaN?size=200x200"
  />
  <div>
    <h2 />
    <p />
    <p>
      City: 
    </p>
  </div>
  <p>
    avatar id - 
    NaN
  </p>
  <button
    onClick={[Function]}
  >
    increase
  </button>
  <button
    onClick={[Function]}
  >
    decrease
  </button>
</div>
`;
